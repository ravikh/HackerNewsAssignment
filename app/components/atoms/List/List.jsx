import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import styles from './List.style';

const StyledList = ({children, type, className}) => 
    React.createElement(
        styled(type)`${styles}`, 
        {type, className}, 
        children
    )  

StyledList.defaultProps = {
    type: 'ul',
    className: ''
}

StyledList.propTypes = {
    type: PropTypes.oneOf(['ul', 'ol']),
    className: PropTypes.string
}

export default StyledList;