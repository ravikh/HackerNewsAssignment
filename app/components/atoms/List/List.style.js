
import {css} from 'styled-components';

const ListStyles = css`
    margin: 0;
    padding: 0;
    list-style: none;

    ${props => {
        return props.type === 'ol' ? 'counter-reset: li;' : ''
    }}
    
`;

export default ListStyles;