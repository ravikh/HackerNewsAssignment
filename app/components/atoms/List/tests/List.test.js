import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import List from '../List';

import Theme from '../../../../theme';

describe('List', () => {
  let list = '';
  const theme = { ...Theme };

  test('should render correctly', () => {
    list = shallow(<List theme={theme}><li></li></List>);
    expect(list).toMatchSnapshot();
  });
});