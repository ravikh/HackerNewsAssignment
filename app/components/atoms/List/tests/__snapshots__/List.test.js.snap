// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
<styled.ul
  className=""
  type="ul"
>
  <li />
</styled.ul>
`;
